common/src/main/java/com/wynntils/mc/render/RenderUtils.java

```diff
@@ -501,6 +501,126 @@ public static void drawTooltip(List<ClientTooltipComponent> lines, PoseStack pos
         poseStack.popPose();
     }
 
+    /** drawProgressBar
+     * Draws a progress bar(textureY1 and textureY2 now specify both textures with background being on top of the bar)
+     *
+     * @param poseStack poseStack to use
+     * @param texture the texture to use
+     * @param x1 left x on screen
+     * @param y1 top y on screen
+     * @param x2 right x on screen
+     * @param y2 bottom right on screen
+     * @param textureX1 texture left x for the part
+     * @param textureY1 texture top y for the part(top of background)
+     * @param textureX2 texture right x for the part
+     * @param textureY2 texture bottom y for the part(bottom of bar)
+     * @param progress progress of the bar, 0.0f to 1.0f is left to right and 0.0f to -1.0f is right to left
+     */
+    public static void drawProgressBar(
+            PoseStack poseStack,
+            Texture texture,
+            float x1,
+            float y1,
+            float x2,
+            float y2,
+            int textureX1,
+            int textureY1,
+            int textureX2,
+            int textureY2,
+            float progress) {
+        int half = (textureY1 + textureY2) / 2;
+        drawProgressBar(poseStack, texture, x1, y1, x2, y2, textureX1, textureY1, textureX2, half + 1, progress, true);
+        drawProgressBar(poseStack, texture, x1, y1, x2, y2, textureX1, half + 1, textureX2, textureY2, progress, false);
+    }
+
+    /** drawProgressBar
+     * Draws a textured progress bar.
+     *
+     * @param poseStack poseStack to use
+     * @param texture the texture to use
+     * @param x1 left x on screen
+     * @param y1 top y on screen
+     * @param x2 right x on screen
+     * @param y2 bottom right on screen
+     * @param textureX1 texture left x for the part
+     * @param textureY1 texture top y for the part
+     * @param textureX2 texture right x for the part
+     * @param textureY2 texture bottom y for the part
+     * @param progress progress of the bar, 0.0f to 1.0f is left to right and 0.0f to -1.0f is right to left
+     * @param background is it drawing the background or not(whole ty part is background)
+     */
+    public static void drawProgressBar(
+            PoseStack poseStack,
+            Texture texture,
+            float x1,
+            float y1,
+            float x2,
+            float y2,
+            int textureX1,
+            int textureY1,
+            int textureX2,
+            int textureY2,
+            float progress,
+            boolean background) {
+        if (background) {
+            Matrix4f matrix = poseStack.last().pose();
+
+            RenderSystem.enableTexture();
+            RenderSystem.setShader(GameRenderer::getPositionTexShader);
+            RenderSystem.setShaderTexture(0, texture.resource());
```

Comment:
Is this what is causing the 
```
 RenderSystem.setShaderTexture(0, GuiComponent.GUI_ICONS_LOCATION)
```
in the mixin?

Can we read out the current shader texture, save it, and restore it, in this function?

Reply:
We cant't. Basically we only have 1 method for rendering healths, food and air. We overwrite the shader in between those render calls, and we need to reset them, since the MC source only sets it before the method call.

Reply:
But we could call `RenderSystem.setShaderTexture(0, GuiComponent.GUI_ICONS_LOCATION)` here instead, and not have it in the mixin, right?

Reply:
We could, but if we use this for anything else (we will) we don't need it.

Reply:
(I'm not sure it's better, though...)

Reply:
I would much rather call it in the mixin than here

Reply:
Ok.